runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -97,6 +97,38 @@ local function tags_lsp_to_vim(diagnostic, client_id)
   return tags
 end
 
+---@param related_informations lsp.DiagnosticRelatedInformation[]
+---@param offset_encoding string
+---@return vim.diagnostic.RelatedInformation[]
+local function related_information_lsp_to_vim(related_informations, offset_encoding)
+  return vim
+    .iter(related_informations or {})
+    :map(
+      --- @param related_information lsp.DiagnosticRelatedInformation
+      --- @return vim.diagnostic.RelatedInformation?
+      function(related_information)
+        local fname = vim.uri_to_fname(related_information.location.uri)
```

Comment:
@gpanders what do you think of making the related_information property a iterable or table?

It would allow us to make this lazy, so that users not using the information wouldn't have to pay for the conversion costs here